risc0/binfmt/src/image.rs

```diff
@@ -224,25 +217,23 @@ impl MemoryImage {
 
     /// Reads the given byte array in this memory image at the given
     /// address  The caller is responsible for ensuring the bytes do
-    /// not overlap a page boundry.
+    /// not overlap a page boundary.
     pub fn load_region_in_page(&self, addr: u32, bytes: &mut [u8]) -> Result<()> {
         let page_idx = self.info.get_page_index(addr);
         let page_start = self.info.get_page_addr(page_idx);
 
-        match self.pages.get(&page_idx) {
-            None => {
-                ensure!(
-                    addr as usize <= MEM_SIZE,
-                    "address {addr:08X} outside MEM_SIZE ({MEM_SIZE:08X})"
-                );
-                bytes.fill(0);
-            }
-            Some(page) => {
-                bytes.clone_from_slice(
-                    &page[(addr - page_start) as usize..(addr - page_start) as usize + bytes.len()],
-                );
-            }
-        };
+        if let Some(page) = self.pages.get(&page_idx) {
+            bytes.clone_from_slice(
+                &page[(addr - page_start) as usize..(addr - page_start) as usize + bytes.len()],
+            );
+        } else {
+            assert!(
```

Comment:
What is the logical difference here if this check is panicked rather than handled as it was before?

Reply:
Oh, you mean from `assert!` to `ensure!`? that's actually a good catch!
Thanks!

Reply:
We were supposed to remove panics from this code section. But while reverting to the preferred style Frank mentioned I make the mistake of pasting the old version verbatim, whereas I should've replaced the panic for an error variant. Thanks for calling that out!

Reply:
Awesome! Wasn't sure if it was a typo or if it was intentional :)